notebooks/page view functional network/page_view_approach_economic_recovery.ipynb

```diff
@@ -0,0 +1,1507 @@
+    "- sum of the shortest page path lengths: pages may be closer to one seed0 node, but far away from the other seed0 node\n",
+    "- between centrality: pages relevant to a WUJ may not neccessarily have many paths that pass through. It is likely that more 'popular' pages will (i.e. ones with higher session hits)\n",
+    "\n",
+    "\n",
+    "Caveat: browse pages are likely to be ranked highly, as these are common pages which may be highly connected to other nodes. Therefore, irrelevant browse pages are removed. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Order by session hits"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sessions = list(sorted(economic_recovery_subgraph, key=lambda x: x[1]['sessionHitsAll'], reverse=True))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Order by shortest page path lengths"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# calculate the paths of the shortest page path\n",
+    "shortest_paths_seed0 = nx.shortest_path_length(g, source='/browse/working/finding-job')\n",
+    "shortest_paths_seed1 = nx.shortest_path_length(g, source='/topic/further-education-skills/apprenticeships')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# seeed0.1: shortest_paths_seed0.1: '/browse/working/finding-job'\n",
+    "shortest_seed0_1 = {k: v for k, v in sorted(shortest_paths_seed0.items(), key=lambda item: item[1], reverse=False)}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# seed0.2: shortest_paths_seed0.2: '/topic/further-education-skills/apprenticeships'\n",
+    "shortest_seed0_2 = {k: v for k, v in sorted(shortest_paths_seed1.items(), key=lambda item: item[1], reverse=False)}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Order by the sum of the shortest page path lengths"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "combined_paths = []\n",
+    "\n",
+    "for page, closeness in finding_job.items():\n",
+    "    for page1, closeness1 in apprentice.items():\n",
+    "        if page == page1: \n",
+    "            combined_score = closeness + closeness1\n",
+    "            combined_paths.append(page)\n",
+    "            combined_paths.append(combined_score)\n",
+    "            \n",
+    "combined_paths=dict(zip(combined_paths[::2], combined_paths[1::2]))\n",
+    "combined_paths={k: v for k, v in sorted(combined_paths.items(), key=lambda item: item[1], reverse=True)}\n",
```

Comment:
I wonder if there is a cleaner way to write this. It looks like you have two dicts: finding_jobs and apprentice. You're creating a new dict, in which you're finding the keys they have in common, then summing the values in each dict, where the keys match. Here is some code that does this in one line.

`set(a) & set(b)` find all the keys in common. Then it's just a case of indexing into the two dicts at the location of those keys in common and adding the values.

```
a = {'a': 3, 'd': 9, 'h': 1}
b = {'a': 1, 'h': 7, 'p': 5}

{k: a[k] + b[k] for k in set(a) & set(b)}
```
Output: `{'a': 4, 'h': 8}`

Reply:
Nice! Refactored code in commit 0ca3b73